Cache/CacheWarmer.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * Copyright 2011 Johannes M. Schmitt <schmittjoh@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace JMS\SerializerBundle\Cache;
+
+use Metadata\MetadataFactoryInterface;
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+
+class CacheWarmer implements CacheWarmerInterface
+{
+    /**
+     * @var array
+     */
+    private $directories = array();
+    /**
+     * @var MetadataFactoryInterface
+     */
+    private $metadataFactory;
+
+    public function __construct(array $directories, MetadataFactoryInterface $metadataFactory)
+    {
+        $this->directories = $directories;
+        $this->metadataFactory = $metadataFactory;
+    }
+
+    public function warmUp($cacheDir)
+    {
+        foreach ($this->directories as $directory) {
+            $files = glob($directory, GLOB_BRACE);
+            foreach ($files as $file) {
+                $className = false; // @todo
```

Comment:
has anybody a nice way to find all the defined classes inside a file? 

checked few answers on stack overflow but all the proposed solutions are "ugly"

Reply:
This is how composer does it: https://github.com/composer/composer/blob/master/src/Composer/Autoload/ClassMapGenerator.php#L120-L214

Reply:
great! a good copy & paste should do the job!